mlflow/gateway/providers/cohere.py

```diff
@@ -174,6 +179,145 @@ def embeddings_to_model(cls, payload, config):
                 )
         return rename_payload_keys(payload, key_mapping)
 
+    @classmethod
+    def chat_to_model(cls, payload, config):
+        if payload["n"] != 1:
+            raise HTTPException(
+                status_code=422,
+                detail=f"Parameter n must be 1 for Cohere chat, got {payload['n']}.",
+            )
+        del payload["n"]
+
+        if "stop" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail="Parameter stop is not supported for Cohere chat.",
+            )
+        payload = cls._scale_temperature(payload)
+
+        messages = payload.pop("messages")
+        last_message = messages.pop()  # pydantic enforces min_items=1
```

Comment:
Do we need to pop out the last message? or can we just use `messages[0]`?

Reply:
in our schema, the last element should be put in `message` field, and the `[0:-1]` ones should be put in `chat_history`. I popped the last element so that `messages` become all chat history.